sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -795,16 +797,24 @@ static const char *getDeviceLibExtensionStr(DeviceLibExt Extension) {
     return "cl_intel_devicelib_imf";
   case DeviceLibExt::cl_intel_devicelib_imf_fp64:
     return "cl_intel_devicelib_imf_fp64";
+  case DeviceLibExt::cl_intel_devicelib_bfloat16:
+    return "cl_intel_bfloat16_conversions";
   }
   throw compile_program_error("Unhandled (new?) device library extension",
                               PI_ERROR_INVALID_OPERATION);
 }
 
 static RT::PiProgram loadDeviceLibFallback(const ContextImplPtr Context,
                                            DeviceLibExt Extension,
-                                           const RT::PiDevice &Device) {
+                                           const RT::PiDevice &Device,
+                                           bool UseNativeLib) {
 
   const char *LibFileName = getDeviceLibFilename(Extension);
+  std::string LibFileNameStr(LibFileName);
+  if (UseNativeLib) {
+    LibFileNameStr.replace(8, 8, "native");
+    LibFileName = LibFileNameStr.c_str();
+  }
```

Comment:
This seems a little elaborate. I am tempted to say we should move the `DeviceLibExt` switch cases (`getDeviceLibExtensionStr` and `getDeviceLibFilename`) to a macro in a .def file (similar to for example sycl/include/sycl/info/device_traits.def) and have both fallback and native in there. Then you can either have `getDeviceLibFilename` pick between native and fallback, or have separate functions and select the right one here.

Reply:
I have changed the code to capture the spirit of your comment. It is not as elaborate as macro definition and uses of that macro in a separate file but does allow a one-line change to add new fallback/native library pairs. Please see if this is acceptable.

Reply:
That is a reasonable alternative. I am okay with that!